File: 1budimas-admin/app/Models/ProdukTipeHarga.php
<?php

namespace App\Models;
use Illuminate\Support\Facades\Session;

class ProdukTipeHarga extends Model
{
    /**
     * Setting Up Intial API Endpoint.
     */
    public function __construct(){
        parent::__construct();
        $this->endpoint = "/api/base/produk_tipe_harga";
    }

    
    /**
     * Insert Data Baru.
     * 
     * @method Override.
     * @param Illuminate\Http\Request|$data Requested HTTP Form Data.
     * @return App\Models\ProdukTipeHarga Instance.
     */
    public function insert($data) {
        return parent::insert([
            'nama' => $data->nama ?? '',
            'kode' => $data->kode ?? ''
        ]);
    }

    /**
     * Update Data Berdasarkan Id.
     * 
     * @param Illuminate\Http\Request|$data Requested HTTP Form Data.
     * @return App\Models\ProdukTipeHarga Instance.
     */
    public function updateById($data) {
        return $this->id($data->id)->update([
            'nama' => $data->nama ?? '',
            'kode' => $data->kode ?? ''
        ]);
    }

    /**
     * Delete Data Berdasarkan Id.
     * 
     * @param Illuminate\Http\Request|$data Requested HTTP Form Data.
     * @return App\Models\ProdukTipeHarga Instance.
     */
    public function deleteById($data) {
        return $this->id($data->id)->delete();
    }
}

File: apps/models/ProdukTipeHarga.py
from apps.models.BaseModel import BaseModel

class ProdukTipeHarga(BaseModel):
    __tablename__ = 'produk_tipe_harga'

    # Kolom tabel
    kode = BaseModel.string(10)
    nama = BaseModel.string(25)

    def __repr__(self):
        return f"ProdukTipeHarga('{self.id}', '{self.nama}')"

    def __init__(self, data=None):
        self.set(data)

    def set(self, data=None):
        if data:
            self.id = data.get('id')
            self.kode = data.get('kode')
            self.nama = data.get('nama')
        return self